pallets/simple-staking/src/lib.rs

```diff
@@ -0,0 +1,242 @@
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Event documentation should end with an array that provides descriptive names for event
+		/// parameters. [something, who]
+		NewInvulnerables(Vec<T::AccountId>),
+		NewAllowedAuthorCount(u32),
+		NewAuthorBond(BalanceOf<T>),
+		AuthorAdded(T::AccountId, BalanceOf<T>),
+		AuthorRemoved(T::AccountId)
+	}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		TooManyAuthors,
+		Unknown,
+		Permission,
+		AlreadyAuthor,
+		NotAuthor
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+	// These functions materialize as "extrinsics", which are often compared to transactions.
+	// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(10_000)]
+		pub fn set_invulnerables(
+			origin: OriginFor<T>,
+			new: Vec<T::AccountId>,
+		) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			if (new.len() as u32) > T::MaxInvulenrables::get() {
+				log::warn!(
+					"invulnerables > T::MaxInvulenrables; you might need to run benchmarks again"
+				);
+			}
+			<Invulnerables<T>>::put(&new);
+			Self::deposit_event(Event::NewInvulnerables(new));
+			Ok(().into())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn set_allowed_author_count(origin: OriginFor<T>, allowed: u32) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			if allowed > T::MaxAuthors::get() {
+				log::warn!(
+					"allowed > T::MaxAuthors; you might need to run benchmarks again"
+				);
+			}
+			<AllowedAuthors<T>>::put(&allowed);
+			Self::deposit_event(Event::NewAllowedAuthorCount(allowed));
+			Ok(().into())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn set_author_bond(origin: OriginFor<T>, author_bond: BalanceOf<T>) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			<AuthorBond<T>>::put(&author_bond);
+			Self::deposit_event(Event::NewAuthorBond(author_bond));
+			Ok(().into())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn register_as_author(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			// lock deposit to start or require min?
+			let who = ensure_signed(origin)?;
+			let deposit = <AuthorBond<T>>::get();
+			let length = <Authors<T>>::decode_len().unwrap_or_default();
+			ensure!((length as u32) < Self::allowed_authors(), Error::<T>::TooManyAuthors);
+			let new_author = AuthorInfo {
+				who: who.clone(),
+				deposit,
+				last_block: None
+			};
+			<Authors<T>>::try_mutate(|authors| -> DispatchResult {
+				if authors.into_iter().any(|author| author.who == who) {
+					Err(Error::<T>::AlreadyAuthor)?
+				} else {
+					T::Currency::reserve(&who, deposit)?;
+					authors.push(new_author);
```

Comment:
@kianenigma - just curious, would it make `into_iter().any()` lighter if this were ordered?

Reply:
We could generally maintain the list of accounts sorted and that would make everything a wee bit faster. But I think for a few dozen invulnerables and at most few tens of bidding authors it is not worth the effort